src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -48,12 +48,12 @@
 from pytorch_lightning.utilities.optimizer import optimizers_to_device
 from pytorch_lightning.utilities.rank_zero import rank_zero_info
 from pytorch_lightning.utilities.seed import reset_seed
-from pytorch_lightning.utilities.types import _PATH, LRSchedulerConfig, LRSchedulerTypeUnion, STEP_OUTPUT
+from pytorch_lightning.utilities.types import _LRScheduler, _PATH, LRSchedulerConfig, ReduceLROnPlateau, STEP_OUTPUT
 from pytorch_lightning.utilities.warnings import rank_zero_warn, WarningCache
 
 warning_cache = WarningCache()
 
-_DEEPSPEED_AVAILABLE: bool = _RequirementAvailable("deepspeed")
+_DEEPSPEED_AVAILABLE: bool = cast(bool, _RequirementAvailable("deepspeed"))
```

Comment:
This should work

```suggestion
_DEEPSPEED_AVAILABLE = _RequirementAvailable("deepspeed")
```

Reply:
It does not work :(. `mypy` throws `error: Cannot determine type of "_DEEPSPEED_AVAILABLE"  [has-type]`.

Reply:
So the error is actually for a different file (`src/pytorch_lightning/plugins/precision/deepspeed.py`). I have applied a quick fix to the said file and this should resolve the problem.